src/fermo/app_utils/input_testing.py

```diff
@@ -0,0 +1,236 @@
+    filename: 'str'
+    file_format: 'str' file format of one specific input with the leading dot
+    allowed_extension: list of strings as specified in config file
+
+    Return
+    ------
+    'None': if the filename is valid
+    'str': otherwise a string with feedback for the user to be flashed
+    '''
+    if '.' not in file_format:
+        raise ValueError('file_format should be specified with the leading dot'
+                         )
+    elif file_format not in allowed_extensions:
+        raise ValueError('file_format must be an allowed_extension as \
+specified in the config')
+    else:
+        if filename.endswith(file_format):
+            return None
+        else:
+            return f"File must be a {file_format}-file!"
+
+
+def save_file(file, file_format, allowed_extensions, upload_folder):
+    '''Save the file if it is valid
+
+    Parameters
+    ----------
+    file: 'werkzeug.datastructures.FileStorage' an element of request.files
+    file_format: 'str' file format of one specific input with the leading dot
+    allowed_extensions: 'list of str' as specified in config file
+    upload_folder: 'str' path to location where files should be stored
+
+    Return
+    ------
+    'str': feedback for the user to be flashed and
+    'str'/'None': filename if user input was valid, or 'None' otherwise
+
+    Notes
+    -----
+    Checks if a file was provided and if it has the right format (judging
+    by the filename). If so, the file is saved in the 'upload_folder'
+    '''
+    filename = secure_filename(file.filename)
+    if filename == '':
+        return 'No file was loaded. Please upload a session-file.', None
+    if file and not check_file_format(filename, file_format,
+                                      allowed_extensions):
+        try:
+            file.save(os.path.join(upload_folder, filename))
+        except FileNotFoundError as e:
+            print(e)
+            return "File or folder didn't exist, so the uploaded file couldn't\
+                be saved", None
+        return 'File loaded successfully', filename
+    else:
+        return f'Upload of {filename} was not successful', None
+
+
+def check_version(content_dict, filename, version):
+    '''Compare version of uploaded file with current FERMO-version
+
+    Parameters
+    ----------
+    content_dict: 'dict' from opened json file
+    filename: 'str'
+    version: 'str' version of the running tool
+
+    Return
+    ------
+    'str': Error/warning message if versions are(may) not (be) compatible,
+    none otherwise
+    '''
+    uploaded_version = content_dict['FERMO_version']
+    uploaded_version_split = uploaded_version.split('.')
+    major, minor, fix = (
+        uploaded_version_split[0],
+        uploaded_version_split[1],
+        uploaded_version_split[2],
+        )
+    if major != version.split('.')[0] or \
+       minor != version.split('.')[1]:
+        return f'''❌ Error: The loaded session file "{filename}"
```

Comment:
Better to use quotes to combine conditionals: 
```
if (
    major != version.split('.')[0] or minor != version.split('.')[1]
):
```

Reply:
(the current behaviour is not adherent to proper semantic versioning  - minor and fix should also be backward compatible - but lets keep it for now)